### ISSUE TITLE
Doing multiple tour

### ISSUE THREAD

**Issue by kevinRoomies** [elrumordelaluz/reactour#439]
I'd like to do multiple tour, one on the first connection and one on a specific page when first accessed (because you might not be able to access it on the first connection).

However if I create a 2nd tour provider it will inevitably be a children of the first one, is there a way to differenciate them when using useTour?

Or would I need to do something like keeping only 1 provider and putting the steps of the 2nd tour at the end of the steps of the first one and manually set the steps based on requirement?

**Comment by elrumordelaluz** [comment]
Why not use one _Provider_ and update `steps` based on _route_?

Something like [this](https://codesandbox.io/s/reactour-tour-demo-using-react-router-dom-kujql?file=/components/Main.js) maybe.

**Comment by eglavin** [comment]
Yeah this was causing me some confusion alright where when I added a another set of steps with less steps then where I have gotten to in the previous usage, The tour wouldn't open or even throw an error. So, I wonder should current step be reset to 0 when setSteps is called?

**Comment by elrumordelaluz** [comment]
> should current step be reset to 0 when setSteps is called?

This depends on your use case. Share a minimal reproduction if you want to analyze together.

**Comment by eglavin** [comment]
Yeah I don't have much of an reproducible example to be honest, I can if required but essentially, I'm upgrading from v1 so I've decided to rewrite the tours from scratch. In doing so, I'm using Reactour for a regular tour and also for a what's new type of tour, which means depending on the button clicked, I'll setSteps with the appropriate set of steps and the setIsOpen true

This all works fine for if you haven't interacted with the the main set of steps, but for the case where there's only a single what's new step and a user has already clicked through to the last step of the main tour, that user is now unable to open the what's new tour due to the currentStep being set to 5 for example. 

Now the reason I commented on this post is due to the fact I only seen in that codesandbox that I should reset the currentStep when changing the steps. So yeah might not be the most important fix but might be a bit more ergonomic then spending ages trying to figure out what happened! 😂😂 That or a bit of information on the docs you have might also help!
